tests/arpack/step-36_parpack_trilinos.cc

```diff
@@ -0,0 +1,342 @@
+
+
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/lac/parpack_solver.h>
+
+#include <fstream>
+#include <iostream>
+
+const unsigned int dim = 2;//run in 2d to save time
+
+using namespace dealii;
+
+const double eps = 1e-10;
+
+template <class DH>
+std::vector<IndexSet>
+locally_owned_dofs_per_subdomain (const DH  &dof_handler)
+{
+  std::vector< types::subdomain_id > subdomain_association (dof_handler.n_dofs ());
+  DoFTools::get_subdomain_association (dof_handler, subdomain_association);
+
+  const unsigned int n_subdomains = 1 + (*max_element (subdomain_association.begin (),
+                                                       subdomain_association.end ()   ));
+
+  std::vector<IndexSet> index_sets (n_subdomains,IndexSet(dof_handler.n_dofs()));
+
+  // loop over subdomain_association and populate IndexSet when a
+  // change in subdomain ID is found
+  types::global_dof_index i_min          = 0;
+  types::global_dof_index this_subdomain = subdomain_association[0];
+
+  for (types::global_dof_index index = 1;
+       index < subdomain_association.size (); ++index)
+    {
+      //found index different from the current one
+      if (subdomain_association[index] != this_subdomain)
+        {
+          index_sets[this_subdomain].add_range (i_min, index);
+          i_min = index;
+          this_subdomain = subdomain_association[index];
+        }
+    }
+
+  // the very last element is of different index
+  if (i_min == subdomain_association.size () - 1)
+    {
+      index_sets[this_subdomain].add_index (i_min);
+    }
+
+  // otherwise there are at least two different indices
+  else
+    {
+      index_sets[this_subdomain].add_range (
+        i_min, subdomain_association.size ());
+    }
+
+  for (unsigned int i = 0; i < n_subdomains; i++)
+    index_sets[i].compress ();
+
+  return index_sets;
+} //locally_owned_dofs_per_subdomain
+
+
+void test ()
+{
+  const unsigned int global_mesh_refinement_steps = 5;
+  const unsigned int number_of_eigenvalues        = 5;
+
+  MPI_Comm mpi_communicator = MPI_COMM_WORLD;
+  const unsigned int n_mpi_processes = Utilities::MPI::n_mpi_processes(mpi_communicator);
+  const unsigned int this_mpi_process = Utilities::MPI::this_mpi_process(mpi_communicator);
+
+
+  Triangulation<dim> triangulation;
+  DoFHandler<dim> dof_handler(triangulation);
+  FE_Q<dim> fe(1);
+  ConstraintMatrix constraints;
+  IndexSet locally_owned_dofs;
+  IndexSet locally_relevant_dofs;
+
+  std::vector<TrilinosWrappers::MPI::Vector> eigenfunctions;
+  std::vector<double>                                eigenvalues;
+  TrilinosWrappers::SparseMatrix             stiffness_matrix, mass_matrix;
+
+  GridGenerator::hyper_cube (triangulation, -1, 1);
+  triangulation.refine_global (global_mesh_refinement_steps);
+
+  //METIS:
+  //GridTools::partition_triangulation (n_mpi_processes, triangulation);
```

Comment:
same here - or convert into comment describing that the following code is a manual partition and mimics above call.